src/robusta/core/sinks/robusta/rrm/prometheus_alert_resource_manager.py

```diff
@@ -0,0 +1,208 @@
+                    "role": "alert-rules",
+                    self.__label_selector_key: self.__label_selector_value
+                },
+            },
+            "spec": {
+                "groups": [
+                    {
+                        "name": self.__group_name,
+                        "rules": rules
+                    }
+                ]
+            }
+        }
+
+    def __create_cr_rules(self, name: str, active_rules: List[PrometheusAlertRule], existing_cr_obj: Optional[dict]) -> \
+            Optional[dict]:
+        try:
+            rules = [rule.to_dict() for rule in active_rules]
+            if existing_cr_obj:
+                if not existing_cr_obj.get("spec"):
+                    existing_cr_obj["spec"] = {}
+
+                if not existing_cr_obj["spec"].get("groups"):
+                    existing_cr_obj["spec"]["groups"] = [{}]
+
+                existing_cr_obj["spec"]["groups"][0]["rules"] = rules
+
+                # If a custom object with the given name already exists then replace the existing custom object with
+                # the updated one.
+                return self.__k8_api.replace_namespaced_custom_object(
+                    group=self.__group,
+                    version=self.__version,
+                    plural=self.__plural,
+                    body=existing_cr_obj,
+                    namespace=self.__installation_namespace,
+                    name=name
+                )
+            else:
+                # If the custom object doesn't exist, create a new one.
+                return self.__k8_api.create_namespaced_custom_object(
+                    group=self.__group,
+                    version=self.__version,
+                    plural=self.__plural,
+                    body=self.__get_snapshot_body(name=name, rules=rules),
+                    namespace=self.__installation_namespace
+                )
+
+        except Exception as e:
+            logging.error(f"An error occured while creating PrometheusRules CRD", exc_info=True)
+
+            raise e
+
+    def __delete_crd_file(self, name: str):
+        try:
+            self.__k8_api.delete_namespaced_custom_object(
+                group=self.__group,
+                version=self.__version,
+                plural=self.__plural,
+                namespace=self.__installation_namespace,
+                name=name,
+                grace_period_seconds=60
+            )
+
+        except Exception as e:
+            logging.error(f"An error occured while deleting the PrometheusRules CRD", exc_info=True)
+
+            raise e
+
+    def start_syncing_alerts(self, account_resources: List[AccountResource]):
+        for res in account_resources:
+            if res.resource_state:
+                rule_dict = res.resource_state.get("rule")
+                rule: PrometheusAlertRule = PrometheusAlertRule.from_supabase_dict(rule_dict)
+
+                in_cluster = self.__in_cluster(res.clusters_target_set)
```

Comment:
As discussed, in the `get_account_resources` in the first polling attempt (when the program starts) we filtering the rows as per account_id, deleted, enabled and clusters_target_set columns. In the subsequent polling we do not use these filters anymore instead we filter by account_id and updated_at . This will reduce the stress on the sass